proto/src/depscloud/api/v1beta/core.proto

```diff
@@ -0,0 +1,53 @@
+//
+// This file attempts to follow common styles and design patterns described by
+// Google Cloud based on their extensive use of protocol buffers.
+//
+// - See here for more: https://cloud.google.com/apis/design/design_patterns
+//
+syntax = "proto3";
+
+package depscloud.api.v1beta;
+
+option go_package = "github.com/depscloud/api/v1beta";
+
+// A source in depscloud represents an origin for information. This includes,
+// but is not limited to, repositories and artifactories.
+message Source {
```

Comment:
Given sources might have different views based on security, should there be some notion of user information here? Could be completely optional.

Reply:
What do you mean by "user"?

Reply:
"Principal" used to connect to the datasource: I'm thinking of a future question about why a module is missing that is answered by "oh, the `build` user cannot see that module."

Reply:
Ah. I see what you mean. i.e. trying to debug why a given source wasn't indexed. I think this has come up a few different ways. One thing I've thought about was having each `account` represented in the graph so you could tell which accounts indexed which sources. This would be one way to understand this as the account would be externally associated with the build user.

We could leverage the `labels` concept below to attach this kind of metadata, I just want to avoid that becoming a sledgehammer.